pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -647,51 +647,78 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
 				}
 			}
 
-			restoreCompleteList = append(restoreCompleteList, existingRestoreVolInfos...)
-			restoreVolumeInfos, err := driver.StartRestore(restore, backupVolInfos, preRestoreObjects)
+			restore, err = a.updateRestoreCRInVolumeStage(
+				namespacedName,
+				storkapi.ApplicationRestoreStatusInProgress,
+				storkapi.ApplicationRestoreStageVolumes,
+				"Volume restores are in progress",
+				existingRestoreVolInfos,
+			)
 			if err != nil {
-				message := fmt.Sprintf("Error starting Application Restore for volumes: %v", err)
-				log.ApplicationRestoreLog(restore).Errorf(message)
-				if _, ok := err.(*volume.ErrStorageProviderBusy); ok {
-					msg := fmt.Sprintf("Volume restores are in progress. Restores are failing for some volumes"+
-						" since the storage provider is busy. Restore will be retried. Error: %v", err)
+				return err
+			}
+			// Get restore volume batch count
+			batchCount := k8sutils.DefaultRestoreVolumeBatchCount
+			restoreVolumeBatchCount, err := k8sutils.GetConfigValue(k8sutils.StorkControllerConfigMapName, metav1.NamespaceSystem, k8sutils.RestoreVolumeBatchCountKey)
+			if err != nil {
+				logrus.Debugf("error in reading %v cm, defaulting restore volume batch count", k8sutils.StorkControllerConfigMapName)
+			} else {
+				if len(restoreVolumeBatchCount) != 0 {
+					batchCount, err = strconv.Atoi(restoreVolumeBatchCount)
```

Comment:
If for some reason configMap value is empty, need to default to count of 25

Reply:
Since batchCount is set to default value at line 661,  it will initialised already.